packages/dashboard/app/withdraw/utils/handlers.ts

```diff
@@ -0,0 +1,245 @@
+  token: TokenBalance,
+  setSelectedToken: (token: TokenBalance | null) => void,
+  setWithdrawAmount: (amount: string) => void
+) => {
+  setSelectedToken(token);
+  setWithdrawAmount('');
+};
+
+export const handleMaxAmount = async (
+  selectedToken: TokenBalance | null,
+  setWithdrawAmount: (amount: string) => void,
+  showStatus: ShowStatusFn
+) => {
+  if (selectedToken) {
+    // If ETH, account for both L1 and L2 fees on Base
+    if (selectedToken.symbol === 'ETH') {
+      try {
+        const gasInfo = await calculateEthGasCosts();
+        // Setting a 100% buffer here for the total cost. Cost is >$.002 and it isn't worth
+        // the potential transaction failure just for the extra $.002
+        const totalCost = gasInfo.totalCost.mul(2);
+
+        if (selectedToken.rawBalance.lte(totalCost)) {
+          setWithdrawAmount('0');
+          showStatus('Insufficient balance for gas fees', 'warning');
+        } else {
+          const maxAmount = selectedToken.rawBalance.sub(totalCost);
+          setWithdrawAmount(ethers.utils.formatEther(maxAmount));
+          showStatus(`Reserved ETH for gas fees`, 'info');
+        }
+      } catch (error) {
+        // Fallback to 80% of balance if estimation fails
+        console.error('Error calculating max amount:', error);
+        const maxAmount = selectedToken.rawBalance.mul(80).div(100);
+        setWithdrawAmount(ethers.utils.formatEther(maxAmount));
+        showStatus('Using fallback 80% of balance due to estimation error', 'warning');
+      }
+    } else {
+      // For tokens, use the full balance
+      setWithdrawAmount(selectedToken.balance);
+    }
+  }
+};
+
+export const handleToggleHideToken = (
+  address: string,
+  e: React.MouseEvent,
+  hiddenTokens: string[],
+  selectedToken: TokenBalance | null,
+  hideToken: (address: string) => void,
+  unhideToken: (address: string) => void,
+  setSelectedToken: (token: TokenBalance | null) => void,
+  setWithdrawAmount: (amount: string) => void
+) => {
+  e.stopPropagation();
+
+  if (hiddenTokens.includes(address.toLowerCase())) {
+    unhideToken(address);
+  } else {
+    hideToken(address);
+
+    // If the hidden token was selected, clear selection
+    if (selectedToken && selectedToken.address.toLowerCase() === address.toLowerCase()) {
+      setSelectedToken(null);
+      setWithdrawAmount('');
+    }
+  }
+};
+
+export const handleRefreshBalances = async (
+  ethAddress: string | undefined,
+  setLoading: (loading: boolean) => void,
+  setBalances: (balances: TokenBalance[]) => void,
+  showStatus: ShowStatusFn,
+  showError: ShowErrorFn
+) => {
+  if (!ethAddress) return;
+
+  try {
+    setLoading(true);
+    const ethBalance = await fetchEthBalance(ethAddress);
+    const erc20TokenBalances = await fetchERC20TokenBalances(ethAddress);
+
+    const balances = [ethBalance.balances[0],...erc20TokenBalances.balances];
+
+    if (erc20TokenBalances.success && ethBalance.success) {
```

Comment:
```suggestion
    if (erc20TokenBalances.success && ethBalance.success) {
        const balances = [...ethBalance.balances, ...erc20TokenBalances.balances];
```
`balances` is only used inside that block. And the existence of `[0]` assumes both `ethBalance.success: true` and that the `ethBalance.balances` array is not empty (just destructuring is looser so it won't throw when the array is empty for example or would allow multiple native balances for when we do this multichain)